expirable_store.go

```diff
@@ -197,9 +219,18 @@ func (s *expirableStore) emptyExpiredBucket() {
 		s.mu.Lock()
 	}
 	defer s.mu.Unlock()
+	entryCount := len(s.buckets[toExpire].entries)
```

Comment:
Any reason this line is above this for loop instead of above the line it is first used at (line 228)?

Reply:
After the for loop `len` should be 0, so we want to know what it was prior to deleting the items.

Reply:
I moved the line down and none of the tests broke. Can we add a test to make sure it isn't moved?

Reply:
Hmm, I'll try to think of a way to address this and [your other comment](https://github.com/hashicorp/go-rate/pull/5#discussion_r1401228381) so that a test can properly assert on this behavior.

Reply:
See the test added in [14ad0e3](https://github.com/hashicorp/go-rate/pull/5/commits/14ad0e369e497749e0ba47f528315eb80e0ee6dd). Now if you move that line down, that test will fail.

Reply:
how about a comment to make it clear "why you can't/shouldn't move this assignment"?

Reply:
Added a comment to explain this.